src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.url;
+    this._headers = options.headers;
+  }
+  _getResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(this._url + "users/me", {
+      headers: this._headers,
+    }).then(this._getResponse);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._getResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.